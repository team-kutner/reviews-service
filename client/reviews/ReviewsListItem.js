// import React, { useState } from 'react';
const {React} = window
const {useState} = React
// import styled, { css } from 'styled-components';
const {styled} = window
const {css} = styled


const ReviewsListItem = (props) => {
  const {modal, isComment=false} = props
  let {author: {username, avatar}, content, createdAt, comments=[]} = props.review
  const date = new window.Date(createdAt)
  const month = date.toLocaleString('default', { month: 'long' })
  const year = date.getFullYear()
  const [readMoreAvail, setReadMoreAvail] = useState(content.length > 180);



  return (
    <Container modal={modal} isComment={isComment}>
      <AvatarContainer>
        <AvatarImage src={avatar}/>
        <UsernameAndDateContainer>
          <Username>{isComment && 'Response from'} {username}</Username>
          <Date>{month} {year}</Date>
        </UsernameAndDateContainer>
      </AvatarContainer>

      <AllContentContainer isComment={isComment}>
        <TextContent>{readMoreAvail ? content.substring(0, 180) + '...' : content}</TextContent>
        {readMoreAvail && <ReadMoreButton onClick={() => setReadMoreAvail(false)}>read more</ReadMoreButton>}
      </AllContentContainer>
      {modal && !isComment && !!comments.length && <ReviewsListItem review={comments[0]} modal={modal} key={comments[0].id} isComment={true}/>}
    </Container>
  );
}

export default ReviewsListItem;

const ReadMoreButton = styled.button`
  font-size: inherit;
  font-style: inherit;
  font-variant: inherit;
  line-height: 24px;
  appearance: none;
  background: transparent;
  border: 0px;
  cursor: pointer;
  margin: 0px;
  padding: 0px;
  user-select: auto;
  color: ${props => props.theme.darkGray};
  text-decoration: underline;
  border-radius: 4px;
  font-weight: ${props => props.theme.mdWeight};
  outline: none;
  display: block;
`
const TextContent = styled.div`
  font-weight: ${props => props.theme.smWeight};
  font-size: ${props => props.theme.mdFont};
  line-height: 24px;
  word-break: break-word;
  color: ${props => props.theme.gray};
`

const AllContentContainer = styled.div`
  ${props => props.isComment && css`
    margin-left: 68px;
  `}

`


const Container = styled.div`
  ${props => !props.modal && css`
    width: 41.66%;
    margin-right: 8.33%;
  `}

  ${props => !props.isComment && css`
    margin-bottom: 40px;
  `}

  ${props => props.isComment && css`
    margin-top: 30px !important;
    margin-left: 32px !important;
  `}
`;

const AvatarContainer = styled.div`
  margin-bottom: 16px;
  display: flex;
  align-items: center;
`
const AvatarImage = styled.img`
  height: 56px;
  width: 56px;
  border-radius: 50%;
`
const UsernameAndDateContainer = styled.div`
  margin-left: 12px;
`
const Username = styled.div`
  color: ${props => props.theme.gray};
  /* font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif; */
  font-weight: ${props => props.theme.mdWeight};
  font-size: ${props => props.theme.mdFont};
  line-height: 20px;
`
const Date = styled.div`
  color: ${props => props.theme.lightGray};
  line-height: 20px;
  font-size: ${props => props.theme.smFont};
  font-weight: ${props => props.theme.smWeight};
`