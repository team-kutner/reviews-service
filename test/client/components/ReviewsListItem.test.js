import Enzyme, {shallow} from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import React from 'react'
import ReviewsListItem from '../../../client/reviews/ReviewsListItem'
import home10ApiResponse from '../../fixtures/home10ApiResponse'

it('should render the first review', () => {
  const wrapper = shallow(<ReviewsListItem review={home10ApiResponse.reviewsWithComments[0]}/>)
  expect(wrapper).toMatchSnapshot()
})

it('should render the first review that has a comment', () => {
  const firstReviewWithComments = home10ApiResponse.reviewsWithComments.find((review) => review.comments.length > 0)

  const wrapper = shallow(<ReviewsListItem review={firstReviewWithComments} modal={true}/>)
  expect(wrapper).toMatchSnapshot()
})
// home10ApiResponse