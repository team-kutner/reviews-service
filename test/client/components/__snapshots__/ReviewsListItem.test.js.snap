// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the first review 1`] = `
<styled.div
  isComment={false}
>
  <styled.div>
    <styled.img
      src="https://s3.amazonaws.com/uifaces/faces/twitter/fabbianz/128.jpg"
    />
    <styled.div>
      <styled.div>
         
        Mittie
      </styled.div>
      <styled.div>
        December
         
        2020
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    isComment={false}
  >
    <styled.div>
      Accusamus eos quia est occaecati rerum. Et amet excepturi voluptatibus in incidunt quibusdam. Velit expedita rerum ea et minima culpa rerum. Rerum aut facilis similique illum repud...
    </styled.div>
    <styled.button
      onClick={[Function]}
    >
      read more
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`should render the first review that has a comment 1`] = `
<styled.div
  isComment={false}
  modal={true}
>
  <styled.div>
    <styled.img
      src="https://s3.amazonaws.com/uifaces/faces/twitter/foczzi/128.jpg"
    />
    <styled.div>
      <styled.div>
         
        Annamarie
      </styled.div>
      <styled.div>
        December
         
        2020
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    isComment={false}
  >
    <styled.div>
      Qui quaerat velit voluptas sit ipsa voluptatem sequi eius distinctio. Repellat incidunt aut. Omnis delectus necessitatibus molestiae voluptates officiis. Dolores aut vero esse quod...
    </styled.div>
    <styled.button
      onClick={[Function]}
    >
      read more
    </styled.button>
  </styled.div>
  <ReviewsListItem
    isComment={true}
    key="5fb45c0bbce68d3c67535fee"
    modal={true}
    review={
      Object {
        "__v": 0,
        "_id": "5fb45c0bbce68d3c67535fee",
        "author": Object {
          "__v": 0,
          "_id": "5fb45c0abce68d3c67535e58",
          "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/nicoleglynn/128.jpg",
          "id": "5fb45c0abce68d3c67535e58",
          "username": "Lexi",
        },
        "content": "Possimus aut aut deleniti nihil odio ut. Sint reiciendis dolor tempora corrupti delectus vel itaque in. Accusamus ut sunt. Omnis eum dolor fugit quo. Animi officia occaecati nesciunt et. Neque voluptatum aperiam in molestiae.",
        "createdAt": "2020-12-01T06:00:00.000Z",
        "id": "5fb45c0bbce68d3c67535fee",
        "review": "5fb45c0abce68d3c67535e73",
      }
    }
  />
</styled.div>
`;
